source/game/kart/CollisionGroup.hh

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include "game/kart/KartParam.hh"
+
+#include <egg/math/Matrix.hh>
+#include <game/field/KCollisionTypes.hh>
+
+#include <array>
+#include <span>
+
+namespace Kart {
+
+struct CollisionData {
+    void reset();
+
+    EGG::Vector3f tangentOff;
+    EGG::Vector3f floorNrm;
+    EGG::Vector3f vel;
+    EGG::Vector3f relPos;
+    EGG::Vector3f movement;
+    f32 speedFactor;
+    f32 rotFactor;
+    Field::KCLTypeMask closestFloorFlags;
+    u32 closestFloorSettings;
+    s32 intensity;
+
+    bool floor;
+};
+
+class Hitbox {
+public:
+    Hitbox();
+    ~Hitbox();
+
+    void update(f32 param_1, f32 totalScale, const EGG::Vector3f &scale, const EGG::Quatf &rot,
+            const EGG::Vector3f &pos);
+
+    void reset();
+    void setScale(f32 scale);
+    void setRadius(f32 radius);
+    void setBspHitbox(const BSP::Hitbox *hitbox);
+    void setWorldPos(const EGG::Vector3f &pos);
+    void setLastPos(const EGG::Vector3f &pos);
+    void setLastPos(const EGG::Vector3f &scale, const EGG::Matrix34f &pose);
+
+    const BSP::Hitbox *bspHitbox() const;
+    EGG::Vector3f worldPos() const;
+    EGG::Vector3f lastPos() const;
+    f32 radius() const;
+
+private:
+    const BSP::Hitbox *m_bspHitbox;
+    f32 m_radius;
+    EGG::Vector3f m_worldPos;
+    EGG::Vector3f m_lastPos;
+    EGG::Vector3f m_relPos;
+};
+
+class CollisionGroup {
```

Comment:
In Ghidra I've seen this referred to as both HitboxGroup and CollisionGroup. I believe I use it a bit interchangeably in this PR, in terms of the class name vs its name as a member variable and the getter functions... Any opinions on which to go with?